src/shared/codelens/localLambdaRunner.ts

```diff
@@ -314,6 +319,40 @@ export class LocalLambdaRunner {
     }
 } // end class LocalLambdaRunner
 
+// TimeoutObject contains an object that can handle both cancellation token- and time limit-style timeout situations.
+export class InvokeTimer {
+    private readonly _startTime: number
+    private readonly _endTime: number
+    private readonly _timer: Promise<boolean>
+    public constructor(timeoutLength: number) {
+        this._startTime = new Date().getTime()
+        this._endTime = this._startTime + timeoutLength
+        this._timer = new Promise<boolean>((resolve) => {
+            setTimeout(() => resolve(false), timeoutLength)
+        })
+    }
+
+    // time limit-style timer
+    // will provide the remaining time left in the timeout to send to other functions
+    public get remainingTime(): number {
+        const curr = new Date().getTime()
+
+        return (this._endTime - curr > 0 ? this._endTime - curr : 0)
+    }
+
+    // cancellation token-style timer
+    // truthy when timer is active, false when timer has ended.
```

Comment:
This doesn't seem right. It will always return a promise, so it is always truthy. If the promise is `await`ed, it will always resolve to `false`.

Reply:
I might look into a different implementation, but that was a bit of the idea behind this. The `await` is used as a time limit, while the non-awaited version is used ti determine whether or not the timer exists.

Reply:
✔️ 
(didn't change the behavior, changed how this function is used and the recommendation)